src/serde_watch.rs

```diff
@@ -22,55 +26,122 @@ pub async fn json_watch<T>(path: impl AsRef<Path>) -> Result<JsonWatch<T>, Error
 where
     T: Clone + DeserializeOwned + Send + Sync + 'static,
 {
-    serde_watch(path, |path, data| {
+    serde_watch::<T, _, StringLoader<T>, _>(path, |path, data| {
         serde_json::from_str(data)
             .map_err(|e| Error::DecodeError(path.to_string_lossy().to_string(), e.to_string()))
     })
     .await
 }
 
-async fn load_file<T, F>(path: impl AsRef<Path>, parse: &F) -> Option<T>
+pub async fn jsonlines_watch<T>(path: impl AsRef<Path>) -> Result<JsonWatch<Vec<T>>, Error>
 where
-    T: Clone,
-    F: (Fn(&Path, &str) -> Result<T, Error>) + Send + Sync + 'static,
+    T: Clone + DeserializeOwned + Send + Sync + 'static,
 {
-    let path = path.as_ref();
-    match fs::read_to_string(path).await {
-        Ok(data) => parse(path, &data)
-            .map_err(|err| error!(?err, ?path, "Error parsing data"))
-            .ok(),
-        Err(err) => {
-            error!(?err, ?path, "Error reading file");
-            None
+    serde_watch::<T, _, LinesLoader<T>, Vec<T>>(path, |path, data| {
+        serde_json::from_str(data)
+            .map_err(|e| Error::DecodeError(path.to_string_lossy().to_string(), e.to_string()))
+    })
+    .await
+}
+
+struct StringLoader<T> {
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<T: Send> Loader<T, T> for StringLoader<T> {
```

Comment:
```suggestion
struct StringLoader {}

impl<T: Send> Loader<T, T> for StringLoader {
```

Reply:
I didn't expect that to compile. Good catch!